### ISSUE TITLE
toast function is not stable, leads to needless re-renders

### ISSUE THREAD

**Issue by pke** [jeanverster/react-native-styled-toast#70]
I was wondering, why my hook that had a dependency to `useToast` was constantly re-rendering, after a toast was emitted.

Turns out, the exported `toast` method from `<ToastProvider>` is not stable!

https://github.com/jeanverster/react-native-styled-toast/blob/develop/src/Context/index.tsx#L51

It needs to be wrapped in `React.useCallback` with a dependency on `maxToasts`.

**Comment by jeanverster** [comment]
Hey @pke - thanks for pointing this out! I will update asap 👌🏼

**Comment by pke** [comment]
@jeanverster I wanted to patch it myself locally for now (using patch-package) but I can't run `yarn build` successfully. I am getting a bunch of those errors:

``` 
src/Box/index.tsx:22:14 - error TS2742: The inferred type of 'systemProps' cannot be named without a reference to 'react-native-styled-toast/node_modules/@types/styled-system'. This is likely not portable. A type annotation is necessary.

22 export const systemProps = compose(
```

**Comment by jeanverster** [comment]
Hey @pke I'm not too sure why that might be happening 🤔 Nevertheless I have made this update, will push a new release in the next 15 minutes or so 👍🏼

**Comment by jeanverster** [comment]
Closed by #72
